options/db_options.cc

```diff
@@ -465,8 +478,24 @@ class DBOptionsConfigurable : public MutableDBConfigurable {
       const ConfigOptions& config_options,
       const std::unordered_map<std::string, std::string>& opts_map,
       std::unordered_map<std::string, std::string>* unused) override {
-    Status s = ConfigurableHelper::ConfigureOptions(config_options, *this,
-                                                    opts_map, unused);
+    // Keep a copy of the registry in case we have to revert it
+    ConfigOptions copy = config_options;
+    std::unordered_map<std::string, std::string> left = opts_map;
+
+    if (copy.registry == immutable_.object_registry) {
+      // The registry from the ConfigOptions matches that of the DB
+      immutable_.object_registry = copy.registry->Clone();
```

Comment:
I'm not very clear about the logic here. If `copy.registry == immutable_.object_registry` why we need 'immutable_.object_registry = copy.registry->Clone()'.

Reply:
Code restructured.  No longer needed